Label/ALAnimatingLabel.m

```diff
@@ -0,0 +1,120 @@
+#import "ALAnimatingLabel.h"
+
+@interface ALAnimatingLabel()
+
+// nonatomic - multi threading and saftey
+// strong how pointer is attached to string
+@property (nonatomic, strong) UILabel *label;
+
+@end
+
+@implementation ALAnimatingLabel
+
+- (instancetype)initWithFrame:(CGRect)frame
+{
+    self = [super initWithFrame: frame];
+    if ( self )
+    {
+        self.label = [[UILabel alloc] initWithFrame: frame];
+        [self addSubview: self.label];
+    }
+    
+    return self;
+}
+
+- (void)setText:(NSString *)text
+{
+    /**
+     Update the UNDERLYING iVar ... basically "@property"s r just syntactic sugar that automatically creates setts and getters for you in the bg...
+     
+     like `@property (nonatomic, copy) NSString *text;`
+     
+     generates the setters and getters for yoU:
+     
+     - (void)setText:(NSString *)text // setter
+     {
+     _text = text;
+     }
+     
+     and
+     
+     - (NSString *)text // getter
+     {
+     return _text;
+     }
+     
+     we override the getter or setter if we want to add some custom functionalnity, like in our case, if the user sets `text` on the `ALAnimatingLabel` instance, we wanna override it to also set the text for our underlyin label.
+     */
+    _text = text;
+    
+    // Set the text for the underlying label
+    self.label.text = text;
+
+    CGFloat expectedWidth = [self calculateExpectedWidth];
+    if ( expectedWidth > self.label.frame.size.width )
```

Comment:
[Read](https://devcards.io/train-wreck) on the train wreck anti-pattern. Apple actually addresses it by introducing convenience methods so you can do:

 ```suggestion
    if ( expectedWidth > CGRectGetWidth(self.label.frame) )
```